.travis.yml

```diff
@@ -4,26 +4,46 @@ jdk:
 install: true
 sudo: false
 # Install mongoDB to perform persistence tests
-services: mongodb
+services:
+  - mongodb
+  - postgresql
+addons:
+  postgresql: 9.6
 cache:
   directories:
-  - "$HOME/.m2"
+  - $HOME/.m2
+  - $HOME/.cache/yarn
 # Install semantic-release
 before_script:
   - yarn global add @conveyal/maven-semantic-release semantic-release
 before_install:
-#- sed -i.bak -e 's|https://nexus.codehaus.org/snapshots/|https://oss.sonatype.org/content/repositories/codehaus-snapshots/|g' ~/.m2/settings.xml
-# set region in AWS config for S3 setup
-- mkdir ~/.aws && printf '%s\n' '[default]' 'aws_access_key_id=foo' 'aws_secret_access_key=bar' 'region=us-east-1' > ~/.aws/config
-- cp configurations/default/server.yml.tmp configurations/default/server.yml
+  #- sed -i.bak -e 's|https://nexus.codehaus.org/snapshots/|https://oss.sonatype.org/content/repositories/codehaus-snapshots/|g' ~/.m2/settings.xml
+  # run a script to see if the e2e tests should be ran
+  - source ./scripts/check-if-e2e-should-run-on-travis.sh
+  # set region in AWS config for S3 setup
+  # add aws credentials for datatools-server
+  - if [ "$SHOULD_RUN_E2E" = "true" ]; then mkdir ~/.aws && printf '%s\n' '[default]' 'aws_access_key_id=${AWS_ACCESS_KEY_ID}' 'aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}' 'region=us-east-1' > ~/.aws/config; else mkdir ~/.aws && printf '%s\n' '[default]' 'aws_access_key_id=foo' 'aws_secret_access_key=bar' 'region=us-east-1' > ~/.aws/config; fi
```

Comment:
I think I tried doing that, but specifically made the script file since it seemed like too much info to put into the .travis.yml file. I added a comment in the .travis.yml that hopefully clarifies things.